pkg/config/fetcher/from_processes.go

```diff
@@ -106,3 +106,24 @@ func ProcessAgentConfig(config config.Reader, getEntireConfig bool) (string, err
 
 	return client.FullConfig()
 }
+
+// DatadogClusterAgentConfig fetch the configuration from the cluster-agent process by querying its HTTPS API
+func DatadogClusterAgentConfig(config config.Reader) (string, error) {
+	err := util.SetAuthToken(config)
+	if err != nil {
+		return "", err
+	}
+
+	port := config.GetInt("cluster_agent.cmd_port")
+	if port <= 0 {
+		return "", fmt.Errorf("invalid cluster_agent.cmd_port -- %d", port)
+	}
+
+	c := util.GetClient()
+	c.Timeout = config.GetDuration("server_timeout") * time.Second
+
+	ipcAddressWithPort := fmt.Sprintf("https://localhost:%d/config", port)
+
+	client := settingshttp.NewClient(c, ipcAddressWithPort, "cluster-agent", settingshttp.NewHTTPClientOptions(util.CloseConnection))
+	return client.FullConfig()
```

Comment:
Could we use `client.FullConfigBySource()` instead to return all the layer of the cluster agent config ? This will make the cluster agent metadata behave in the same ways than other. If you're using the default config comp this is done out of the box for you.

We could do this in another PR if needed.